libs/base/Data/List/HasLength.idr

```diff
@@ -12,8 +12,8 @@
 |||
 ||| We would write either of:
 ||| ```idris example
-||| f1 : (n : Nat) -> (0 _ : HasLength xs n) -> P xs
-||| f2 : (n : Subset n (HasLength xs)) -> P xs
+||| f1 : (n : Nat) -> (0 _ : HasLength n xs) -> P xs
+||| f2 : (n : Subset n (flip HasLength xs)) -> P xs
```

Comment:
I'd say it can be written ``(`HasLength` xs)`` but the punning does not work anymore with the new argument order. :sweat_smile:

Reply:
It risks being obvious, but "xs does have a length" ;)

Reply:
Oh, I think I see what you mean. `xs HasLength n` used to read well. 

Maybe `HasLength n xs` should have been named `IsLength` to remain reminiscent of English. I wouldn't vote for renaming now, though.